src/vs/editor/common/services/modelServiceImpl.ts

```diff
@@ -617,6 +617,7 @@ export class ModelServiceImpl extends Disposable implements IModelService {
 	}
 
 	private _onDidChangeLanguage(model: ITextModel, e: IModelLanguageChangedEvent): void {
+		model.removeAllDecorationsWithOwnerId(0);
```

Comment:
So why is this required for our editors? This issue doesn't seem to happen for VS Code editors so it'd be good to find out why that is before we start changing their code.

(and FYI - in the future if we ever make changes to code under the vs folder you should be adding a `{{ SQL CARBON EDIT }}` comment to the lines you're changing so we can track that when doing merges.